### ISSUE TITLE
Will KIP-974 obviate the need for this project?

### ISSUE THREAD

**Issue by k-wall** [ozangunalp/kafka-native#229]
I've really appreciated the trail blazing this project has done in helping make native Kafka a thing in the community.

Now that [KIP-974](https://cwiki.apache.org/confluence/display/KAFKA/KIP-974%3A+Docker+Image+for+GraalVM+based+Native+Kafka+Broker) has materialised and Apache Kafka publishes its own native images, how much value does this project continue to have? Should it sunset?

I can think of some use-cases that this project provides that aren't covered by KIP-974.

* this project provides a simplified interface for starting a broker.
* this project includes some Strimzi SASL plugins that aren't included in Apache Kafka native.
* this project provides a native Zookeeper which is valuable for those testing pre-KRaft use-cases.

Are there others?

**Comment by ozangunalp** [comment]
I agree with this. I've been trying to find the time to look into this. I can add more into the pro-list:

* Build-time init approach brought by Quarkus allows consuming less memory at startup. When compared the RSS of this image is 3 times less than the official kafka-native image. It would be interesting to bench this more and contribute some build time init back to upstream Kafka native build.

As for the cons : 
* Maintenance burden, with the coming 4.0 release
* Image composition doesn't follow standard Apache Kafka distributions. For example, people have been looking for some scripts to run #192, #216

**Comment by k-wall** [comment]
Yes, it would be fantastic to see those RSS improvements back in upstream kafka's native image.  I wish I could say I could help, but it is not really part of my skill set.  If there's a break down of tasks, I might see if I can help with some of the lifting.

I agree with the cons you've highlighted.  Having an Kafka image with a well known command line interface so really going to help the community innovate.  So I think long term having ozangunalp/kafka-native out there too with a different API probably isn't too helpful.

That said, it would really help the Kroxylicious project if this project could produce a  Kafka 3.9 image (when 3.9 comes out). We'll probably use the point Kafka 4.0 is released as the trigger to move to testing with Apache Kafka images.

**Comment by ozangunalp** [comment]
> That said, it would really help the Kroxylicious project if this project could produce a Kafka 3.9 image (when 3.9 comes out). We'll probably use the point Kafka 4.0 is released as the trigger to move to testing with Apache Kafka images.

That's a nice plan.

**Comment by k-wall** [comment]
We at the Kroxylicious project were talking about Kafka 4.0 arrival ([RC4 voting thread)](https://lists.apache.org/thread/hr0l48rf172ghrcolwv3zpq5bcrmnjnt).   On reflection we think it would be a gentler transition if this project supported Kafka 4.0.  I'm intended to open a PR soon with Kafka 4.0 support.  It'll strip out all the Zookeeper support that exists at the moment.

Does that sound okay?

Kroxylicious is still intending to move to use Apache Kafka images at some point in the future, but we want to decouple it from Apache Kafka 4.0's arrival.
